impl/src/test/java/auth/JwtAuthenticatorTest.java

```diff
@@ -9,24 +11,30 @@
 
 import java.time.OffsetDateTime;
 import java.time.ZoneId;
-import java.time.ZoneOffset;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public class JwtAuthenticatorTest {
 	private static final String JWT = "eyJhbGciOiJIUzI1NiIsImtpZCI6ImIxNWEyYjhmN2E2YjNmNmJjMDhiYzFjNTZhODg0MTBlMTQ2ZDAxZmQiLCJ0eXAiOiJKV1QifQ.eyJlbWFpbCI6ImplcHAzX2RyYWdvbnNsYXllckBmb3J0bm94LnNlIiwibmFtZSI6ImplcHAzIiwicGljdHVyZSI6InVybHRvcGljdHVyZSIsInVzZXJfaWQiOjEsImlhdCI6MTU0ODQxNzY0NywiZXhwIjoxNTQ4NDIxMjQ3fQ.yDPCbMe5ZPqNubrWBoJJytk3DqS5FiEVotirtj3wzNA";
 	private JwtAuthResolver jwtAuthResolver;
-	private DateProvider dateProvider;
+    private ApplicationTokenVerifier applicationTokenVerifier;
+
 	@Before
 	public void beforeEach() {
-		dateProvider = mock(DateProvider.class);
+        DateProvider dateProvider = mock(DateProvider.class);
+        applicationTokenVerifier = mock(ApplicationTokenVerifier.class);
+
+        DecodedJWT decodedJWT = com.auth0.jwt.JWT.decode(JWT);
+        when(applicationTokenVerifier.verifyAndDecode(any())).thenReturn(decodedJWT);
+
 		final String pastTime = "2019-01-25T13:00:47+01:00";
 		when(dateProvider.getOffsetDateTime()).thenReturn(OffsetDateTime.parse(pastTime));
 		when(dateProvider.getDefaultZone()).thenReturn(ZoneId.of("Z"));
```

Comment:
Maybe promote the value of DateProvider.getDefaultZone to a static field in DateProviderImpl if getDefaultZone is called often.